etcdserver/api/membership/cluster.go

```diff
@@ -279,12 +286,21 @@ func (c *RaftCluster) ValidateConfigurationChange(cc raftpb.ConfChange) error {
 				plog.Panicf("unmarshal member should never fail: %v", err)
 			}
 		}
-		for _, u := range m.PeerURLs {
-			if urls[u] {
-				return ErrPeerURLexists
+
+		if members[id] != nil && members[id].IsLearner && cc.Type == raftpb.ConfChangeAddNode {
```

Comment:
Can we reuse the already defined `IsPromoteChange` function?

Reply:
`IsPromoteChange` encapsulates `the members, _ := membersFromStore(c.lg, c.v2store)` . If we use the same logic. we must call membersFromStore again to get `members`.